litmus-portal/graphql-server/pkg/chaoshub/chaoshub.go

```diff
@@ -515,31 +513,37 @@ func DeleteChaosHub(ctx context.Context, hubID string, projectID string) (bool,
 }
 
 // GetIconHandler ...
-var GetIconHandler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-	vars := mux.Vars(r)
-	var img *os.File
-	var err error
-	var responseStatusCode int
-	if strings.ToLower(vars["ChartName"]) == "predefined" {
-		img, err = os.Open("/tmp/version/" + vars["ProjectID"] + "/" + vars["HubName"] + "/workflows/icons/" + vars["IconName"])
+var GetIconHandler = gin.HandlerFunc(func(c *gin.Context) {
+	var (
+		projectID          = c.Param("ProjectID")
+		hubName            = c.Param("HubName")
+		chartName          = c.Param("ChartName")
+		iconName           = c.Param("IconName")
+		img                *os.File
+		err                error
+		responseStatusCode int
+	)
+
+	if strings.ToLower(chartName) == "predefined" {
+		img, err = os.Open("/tmp/version/" + projectID + "/" + hubName + "/workflows/icons/" + iconName)
```

Comment:
I mentioned this code block again because of an outdated issue. @S-ayanide @amityt 

@S-ayanide said,
We might also need to support user defined vars, @amityt can you please take a look here!

Reply:
Few validations are required for user provided inputs, we can add it in a separate PR.